### DISCUSSION TITLE
Error: Missing lock for job

### DISCUSSION THREAD

**Discussion by mxm199** [immich-app/immich#13541] [Q&A]
### The bug

Maybe there is a way to reset the locks, restarting the containers did not help

### The OS that Immich Server is running on

Windows Desktop Docker

### Version of Immich Server

1.118.1

### Version of Immich Mobile App

1.118.1

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Standart
```

### Your .env content

```Shell
Standart
```

### Reproduction steps

1. After clicking "Refresh faces" while viewing a photo
2. After editing or setting the location in the photo

### Relevant log output

```shell
immich-server            | Error: Missing lock for job 1. failed
immich-server            |     at Scripts.finishedErrors (/usr/src/app/node_modules/bullmq/dist/cjs/classes/scripts.js:266:24)
immich-server            |     at Job.moveToFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/job.js:427:32)
immich-server            |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
immich-server            |     at async handleFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:379:21)
immich-server            |     at async Worker.retryIfFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:581:24)
immich-server            | Error: Missing lock for job 2. failed
immich-server            |     at Scripts.finishedErrors (/usr/src/app/node_modules/bullmq/dist/cjs/classes/scripts.js:266:24)
immich-server            |     at Job.moveToFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/job.js:427:32)
immich-server            |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
immich-server            |     at async handleFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:379:21)
immich-server            |     at async Worker.retryIfFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:581:24)
immich-server            | Error: Missing lock for job 3. failed
immich-server            |     at Scripts.finishedErrors (/usr/src/app/node_modules/bullmq/dist/cjs/classes/scripts.js:266:24)
immich-server            |     at Job.moveToFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/job.js:427:32)
immich-server            |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
immich-server            |     at async handleFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:379:21)
immich-server            |     at async Worker.retryIfFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:581:24)
```

**Comment by bo0tzz** [comment]
Try fully recreating the containers. `docker compose down && docker compose up -d`

**Comment by mxm199** [comment]
The errors went away after resetting the settings in the Job Settings (concurrency)

> The 'missing lock for job' error in BullMQ generally happens when a process tries to fetch a job from the queue which has already been locked by another process. It could be due to concurrency issues or when multiple workers are trying to access the same job simultaneously.
This can also happen if a worker process crashes or is terminated unexpectedly, leaving a job locked without releasing it. When another worker tries to access this job, the system will throw a 'missing lock for job' error because the lock created by the crashed/terminated process still exists.

**Comment by mxm199** [comment]
Hmm, I was in a hurry... The errors went away when -

After clicking "Refresh faces" while viewing a photo
After editing or setting the location in the photo

But they remained when importing images (probably when processing/receiving metadata/detecting faces)

I tried to set all the concurrency settings to 1, but it doesn't help

> Try fully recreating the containers. `docker compose down && docker compose up -d`

Of course I restarted it like that

**Comment by mertalev** [comment]
Can you check CPU and IO utilization? The "missing lock" issue can also happen when the server is being stressed out too much and can't refresh the locks for its jobs.
